hexitor/src/hex_ctl.cpp

```diff
@@ -168,15 +168,55 @@ void hex_ctl::update(const UINT64 offset, const vector<BYTE>& ori_data, const ma
 	}
 
 	//Highlight cursor position
-	const COORD pos_hex = cursor_from_offset(offset, cursor, true);
-	write(pos_hex.Y, pos_hex.X + 0, settings::clr_active);
-	write(pos_hex.Y, pos_hex.X + 1, settings::clr_active);
-	if (_codepage == CP_UTF16LE && !hex_area) {
-		write(pos_hex.Y, pos_hex.X + 3, settings::clr_active);
-		write(pos_hex.Y, pos_hex.X + 4, settings::clr_active);
+	if (!hex_area && _codepage == CP_UTF8)
+	{
+		// UTF-8 multi-byte character highlighting
+		UINT64 start_char_offset = cursor;
+		size_t rel_offset = static_cast<size_t>(start_char_offset - offset);
+
+		// Find the beginning of the character by moving backwards
+		while (rel_offset > 0 && (ori_data[rel_offset] & 0xC0) == 0x80) {
+			start_char_offset--;
+			rel_offset--;
+		}
+
+		// Determine character length
+		BYTE first_byte = (start_char_offset >= offset) ? ori_data[rel_offset] : 0;
+		if (upd_data.count(start_char_offset)) {
+			first_byte = upd_data.at(start_char_offset);
+		}
+		
+		int char_len = 1;
```

Comment:
why not call get_utf8_char_len instead of this copypaste?

Reply:
fixed